include/beman/optional26/detail/iterator.hpp

```diff
@@ -43,18 +43,20 @@ struct contiguous_iterator : public base_contiguous_iterator<T, Container> {
     using typename base_type::reference;
 
     // Default constructor.
-    contiguous_iterator() noexcept : m_current() {}
+    constexpr contiguous_iterator() noexcept : m_current() {}
 
     // Pointer to iterator constructor.
-    contiguous_iterator(pointer it) noexcept : m_current(it) {}
+    constexpr contiguous_iterator(pointer it) noexcept : m_current(it) {}
```

Comment:
This ctor currently allows implicit conversion, which in turn allows more permissive operations betweeen `contiguous_iterator` and raw pointer.
Should we disallow such implicit conversion, or even furtherly disallow explicit construction? (Perhaps should do in another PR.)

Reply:
I've got a branch with updated wording and a number of fixes to conversion and dangling issues. It's exactly the current proposal. Getting the constraints and mandates to all play nice was a little tricky, but we did notice that assignment boiled down to one assignment from tuple, and that the move and copy assignments were identical, so we only need one. 

It is using the newish reference_converts_from_temporary the way that tuple does now. 

PR will be up later today.

Reply:
The branch currently using aliases the traits that it uses into the Optional26 namespace in order to get the wording synopsis, constraints, mandates, effects all correct and in a form that I can just paste into the TeX doc for standardization, but probably shouldn't be done for the production version. 

It's also based on the pre renaming version. 

The really good news is that an LWG expert helped get it right, so this should be much smoother in the LEWG approval of wording and actual LWG review.